Libraries/Opc.Ua.Server/Subscription/MonitoredItem.cs

```diff
@@ -303,6 +321,14 @@ public bool IsReadyToTrigger
             }
         }
 
+        /// <summary>
+        /// Setup the resend data trigger by setting the monitor item in ResendData state
+        /// </summary>
+        public void SetupResendDataTrigger()
+        {
+            int originalValue = Interlocked.CompareExchange(ref m_resendDataState, (int)ResendDataState.ResendData, (int)ResendDataState.NonResendData);
+        }
```

Comment:
I think a simple Exchange would do it here. Have you checked if you could use Exchange<T> where T could be a bool?

Reply:
Not sure the Interlocked operation is even necessary here. 
The CLI standard (https://www.ecma-international.org/publications-and-standards/standards/ecma-335/) states:

"A conforming CLI shall guarantee that read and write access to properly aligned memory
locations no larger than the native word size (the size of type native int) is atomic
(see §I.12.6.2) when all the write accesses to a location are the same size"